homepages/homepage.php

```diff
@@ -5,23 +5,25 @@
  */
 function largo_register_default_homepage_layouts() {
 	// Load layouts from `layouts/`
-	$layouts = glob(__DIR__ . '/layouts/*.php');
-	foreach ($layouts as $layout)
-		include_once $layout;
-
-	// Load zone components from `zones/`
-	$zones = glob(__DIR__ . '/zones/*.php');
-	foreach ($zones as $zone)
-		include_once $zone;
-
 	$default_layouts = array(
 		'HomepageBlog',
 		'HomepageSingle',
 		'HomepageSingleWithFeatured',
 		'HomepageSingleWithSeriesStories',
 		'TopStories',
-		'LegacyThreeColumn'
+		'LegacyThreeColumn',
 	);
+	foreach( $default_layouts as $layout ) {
+		$layout_file = get_stylesheet_directory() . '/homepages/layouts/' . $layout . '.php';
+		if ( 0 === validate_file( $layout_file ) ) {
+			require_once( $layout_file );
+		}
+	}
+
+	// Load zone components from `zones/`
+	$zones = glob(__DIR__ . '/zones/*.php');
```

Comment:
Can you fix this glob as well?

Reply:
Combined with Layouts and minor refactor too.